src/Checkbox.tsx

```diff
@@ -0,0 +1,62 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement, useLayoutEffect, useRef} from 'react'
+import sx, {SxProp} from './sx'
+import {COMMON} from './constants'
+
+export type CheckboxProps = {
+  /**
+   * Apply indeterminate visual appearance to the checkbox
+   */
+  indeterminate?: boolean
+  /**
+   * Apply inactive visual appearance to the checkbox
+   */
+  disabled?: boolean
+
+  /**
+   * Forward a ref to the underlying input element
+   */
+  ref?: React.RefObject<HTMLInputElement>
+} & InputHTMLAttributes<HTMLInputElement> &
+  SxProp
+
+const StyledCheckbox = styled.input`
+  cursor: pointer;
+
+  ${props => props.disabled && `cursor: not-allowed;`}
+
+  ${COMMON};
+  ${sx}
+`
+
+/**
+ * An accessible, native checkbox component
+ */
+const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
```

Comment:
Why do we need to use `forwardRef`?

Reply:
It's there to make the checkbox more flexible when used in uncontrolled mode.